plothist/plotters.py

```diff
@@ -291,18 +291,121 @@ def plot_2d_hist(hist, ax, pcolormesh_kwargs={}, colorbar_kwargs={}):
         The 2D histogram to plot.
     ax : matplotlib.axes.Axes
         The Axes instance for plotting.
+    ax_colorbar : matplotlib.axes.Axes or None, optional
+        The Axes instance for the colorbar. If None, the colorbar is appended to ax. Default is None.
     pcolormesh_kwargs : dict, optional
         Additional keyword arguments forwarded to ax.pcolormesh() (default is {}).
     colorbar_kwargs : dict, optional
         Additional keyword arguments forwarded to ax.get_figure().colorbar() (default is {}).
     """
-    if "edgecolors" not in pcolormesh_kwargs.keys():
-        pcolormesh_kwargs["edgecolors"] = "face"
+    if ax_colorbar is None:
+        divider = make_axes_locatable(ax)
+        ax_colorbar = divider.append_axes("right", size="5%", pad=0.05)
+    pcolormesh_kwargs.setdefault("edgecolors", "face")
     im = ax.pcolormesh(*hist.axes.edges.T, hist.values().T, **pcolormesh_kwargs)
-    divider = make_axes_locatable(ax)
-    cax = divider.append_axes("right", size="5%", pad=0.05)
-    ax.tick_params(axis="x", which="both", top=False, bottom=False)
-    ax.get_figure().colorbar(im, cax=cax, **colorbar_kwargs)
+    ax.get_figure().colorbar(im, cax=ax_colorbar, **colorbar_kwargs)
+
+
+def plot_2d_hist_with_projections(
+    hist,
+    xlabel=None,
+    ylabel=None,
+    ylabel_x_projection=None,
+    colorbar_label=None,
+    save_as=None,
+    pcolormesh_kwargs={},
+    colorbar_kwargs={},
+):
+    """Plot a 2D histogram with projections on the x and y axes.
+
+    Parameters
+    ----------
+    hist : 2D boost_histogram.Histogram
+        The 2D histogram to plot.
+    xlabel : str, optional
+        Label for the x axis. Default is None.
+    ylabel : str, optional
+        Label for the y axis. Default is None.
+    ylabel_x_projection : str, optional
+        Label for the y label of the x projection. Default is None.
+    colorbar_label : str, optional
+        Label for the colorbar. Default is None.
+    save_as : str, optional
+        Path to save the figure to. Default is None.
+    pcolormesh_kwargs : dict, optional
+        Keyword arguments for the pcolormesh call. Default is {}.
+    colorbar_kwargs : dict, optional
+        Keyword arguments for the colorbar call. Default is {}.
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        The figure.
+    ax_2d : matplotlib.axes.Axes
+        The axes for the 2D histogram.
+    ax_x_projection : matplotlib.axes.Axes
+        The axes for the x projection.
+    ax_y_projection : matplotlib.axes.Axes
+        The axes for the y projection.
+    ax_colorbar : matplotlib.axes.Axes
+        The axes for the colorbar.
+    """
+
+    colorbar_kwargs.setdefault("label", colorbar_label)
+
+    fig, axes = plt.subplots(
+        figsize=(5.25, 5),
```

Comment:
Why this figsize? I find the projection plots to be a bit too squeezed. Could we increase their height a bit?